django/university/routes/MarketplaceExchangeView.py

```diff
@@ -0,0 +1,186 @@
+        if course_unit_name_filter:
+            marketplace_exchanges = list(filter(
+                lambda x: self.courseUnitNameFilterInExchangeOptions(x.options, course_unit_name_filter),
+                marketplace_exchanges
+            ))
+
+        return self.build_pagination_payload(request, marketplace_exchanges)
+
+    def advanced_classes_filter(self, marketplace_exchanges, classes_filter):
+        filtered_marketplace_exchanges = []
+        for exchange in marketplace_exchanges:
+            exchange_tainted = False
+            for option in exchange.options:
+                if option.course_unit_acronym in classes_filter.keys():
+                    correct_class_included = option.class_issuer_goes_from in classes_filter[option.course_unit_acronym]
+                    exchange_tainted = not correct_class_included
+
+            if not exchange_tainted:
+                filtered_marketplace_exchanges.append(exchange)
+
+        return filtered_marketplace_exchanges
+
+    def courseUnitNameFilterInExchangeOptions(self, options, courseUnitNameFilter):
+        matches = []
+        for courseUnitId in courseUnitNameFilter:
+            for option in options:
+                if courseUnitId == option.course_unit_id:
+                    matches.append(1)
+
+        return len(matches) == len(courseUnitNameFilter)
+   
+
+    """
+        Returns all the current marketplace exchange requests paginated
+    """
+    def get(self, request):
+        courseUnitNameFilter = request.query_params.get('courseUnitNameFilter', None)
+        classesFilter = self.parseClassesFilter(request.query_params.get('classesFilter', None))
+    
+        return JsonResponse(self.filterAllExchanges(request, courseUnitNameFilter.split(',') if courseUnitNameFilter else None, classesFilter), safe=False)
+
+    def parseClassesFilter(self, classesFilter: str) -> dict:
+        if not classesFilter:
+            return {}
+
+        b64_decoded = base64.b64decode(classesFilter)
+        string = b64_decoded.decode('utf-8')
+        return dict(json.loads(string))
+
+    def post(self, request):
+        return self.submit_marketplace_exchange_request(request)
+
+    def getExchangeOptionClasses(self, options):
+        classes = sum(list(map(lambda option: ClassController.get_classes(option.course_unit_id), options)), [])
+        return filter(lambda currentClass: any(currentClass["name"] == option.class_issuer_goes_from for option in options), classes)
+
+    def submit_marketplace_exchange_request(self, request):
+        exchanges = request.POST.getlist('exchangeChoices[]')
+        exchanges = list(map(lambda exchange : json.loads(exchange), exchanges))
+
+        print("Marketplace exchange: ", exchanges)
+        curr_student = request.user.username
+        sigarra_res = SigarraController().get_student_schedule(curr_student)
+        
+        if(sigarra_res.status_code != 200):
+            return HttpResponse(status=sigarra_res.status_code)
```

Comment:
Shouldn't we validate the exchanges?

Reply:
I read this part of code and the verification is done after the if on line 145.

It only enters inside the if the the status code of getting the student schedule is not OK (200).